### PR TITLE
Enabling the build for rcl_logging_log4cxx

### PR DESCRIPTION
**Pull request by nburek**
This review includes all the fixes needed to build and run the rcl_logging_log4cxx package on Linux, Mac, and Windows 10. In order for this to work in the build system we will need use brew to install log4cxx on Mac and use the choco package in the related PR in order to install log4cxx on Windows. 

There is a known issue with the Windows build that causes log4cxx to crash when compiled as a Release build, but works if it is compiled in Debug mode. As a temporary work around we force the Windows build of rcl_logging_log4cxx to use Debug mode. 

Requires ros2/choco-packages#7

### CODE REVIEW

**Comment by mjcarroll** [comment]
With the choco package in, this should work now?

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/6227/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/2668/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/5128/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/6072/)

**Comment by mjcarroll** [comment]
After installing choco packages on Windows (via ros2/build_cop#164)

* Windows [Build Status](http://ci.ros2.org/job/ci_windows/6075/)

**Comment by mjcarroll** [comment]
Rebuild with log4cxx installed on macOS and Windows, subset of packages.

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/6231/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/2672/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/5132/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/6077/)

**Comment by mjcarroll** [comment]
Alright @nburek looks mostly good, with a few warnings on Windows CI.  Mind getting those cleaned up?

**Comment by mjcarroll** [comment]
Also, can you add a PR for installation documentation (https://github.com/ros2/ros2_documentation/tree/master/source/Installation) to show that

1. We need to install the `log4cxx` choco package on Windows source/binary
2. We need to install `log4cxx` via brew on macOS.

**Comment by nburek** [comment]
Someone from our team is starting to fix the windows warning and I'm working on adding the documentation to the install instructions. 

Will update as soon as we're ready. Thanks.

**Comment by nburek** [comment]
Documentation updated here: https://github.com/ros2/ros2_documentation/pull/127

We are still working on cleaning up the warnings.

**Comment by juanrh** [comment]
All windows warnings should be solved now. I have disabled [C4275](https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-2-c4275?view=vs-2017) and [C4251](https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-1-c4251?view=vs-2017) because, as discussed [here](https://web.archive.org/web/20130317015847/http://connect.microsoft.com/VisualStudio/feedback/details/696593/vc-10-vs-2010-basic-string-exports) 

> I recommend avoiding this in the first place - putting STL types in your DLL's interface forces you to play by the STL's rules (specifically, you can't mix different major versions of VC, and your IDL settings must match). However, there is a workaround. C4251 is essentially noise and can be silenced...

and those warnings happen on `Log4cxx`'s code. I have added a comment about that in `CMakeLists.txt` next to where those warnings are disabled.

**Comment by nburek** [comment]
This will need the CI build run again to show the Windows warnings were fixed before merge. 

Thanks.

**Comment by mjcarroll** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/6259/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/2686/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/5145/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/6095/)

File: rcl_logging_log4cxx/CMakeLists.txt [hunk 1]
```diff
@@ -12,35 +12,45 @@ if(NOT CMAKE_CXX_STANDARD)
 endif()
 
 find_package(ament_cmake_ros REQUIRED)
+list(INSERT CMAKE_MODULE_PATH 0 "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+find_package(Log4cxx REQUIRED)
 
-set(dependencies
-    log4cxx
-)
+include_directories(${ament_INCLUDE_DIRS} include)
```
Comment (dirk-thomas) [inline rcl_logging_log4cxx/CMakeLists.txt:18]: The variable `ament_INCLUDE_DIRS` isn't defined. Removed in #30.

**Pull request merged**